### ISSUE TITLE
Loading a HEIC image in JavaCV

### ISSUE THREAD

**Issue by Draczech** [bytedeco/javacv#2298]
Hi, I'm trying to load a HEIC image (just a random photo taken by an iPhone) in my project. 

I'm using following setup:
- linux-x86_64 platform, ubuntu jammy or noble LTS
- Java 11 with gradle 8.10.2
- javacv 1.5.10 + opencv 4.9.0 (basically latest releases)

Sample of my code. `ImageType` is just an enum.

```java
public class OpencvVideoDecoder implements ImageDecoder {

  /** Lazy initialized because of serialization. */
  private transient OpenCVFrameConverter.ToMat cachedMatConverter;
  /** Lazy initialized because of serialization. */
  private transient Java2DFrameConverter cachedJavaFrameConverter;

  @Override
  public Mat decode(ImageType imageType, byte[] image) throws ImageDecodingException {
    FFmpegLogCallback.set();
    try (FFmpegFrameGrabber grabber = new FFmpegFrameGrabber(new ByteArrayInputStream(image))) {

      grabber.start();

      // Grab the first frame of the file
      Frame frame = grabber.grabImage();
      if (frame != null) {
        BufferedImage bufferedImage = getJavaFrameConverter().convert(frame);
        return getMatConverter()
                .convertToMat(getJavaFrameConverter().getFrame(bufferedImage))
                .clone();
      } else {
        throw new ImageDecodingException("No Frame found.");
      }
    } catch (Exception e) {
      throw new ImageDecodingException("Unable to decode image", e);
    }
  }

  private synchronized OpenCVFrameConverter.ToMat getMatConverter() {
    if (cachedMatConverter == null) {
      cachedMatConverter = new OpenCVFrameConverter.ToMat();
    }
    return cachedMatConverter;
  }

  private synchronized Java2DFrameConverter getJavaFrameConverter() {
    if (cachedJavaFrameConverter == null) {
      cachedJavaFrameConverter = new Java2DFrameConverter();
    }
    return cachedJavaFrameConverter;
  }

  @Override
  public Collection<ImageType> getSupportedImageTypes() {
    return ImmutableList.of(ImageType.AVIF, ImageType.HEIC);
  }
}
```

The code described above works for avif files, but not for heic. It seems to me, that `implementation "org.bytedeco:ffmpeg:${ffmpegVersion}:linux-x86_64"` dependency cannot work properly with HEIC format. I'm getting `Error: [mov,mp4,m4a,3gp,3g2,mj2 @ 0x72abc928e8c0] moov atom not found` in my code.

While in my machine setup I was able to overcome that issue by installing some additional ubuntu packages. Can HEIC be loaded with current version of `org.bytedeco:ffmpeg`? Do I need some additional dependencies to add to my project? (ideally precompiled)

Alternatively is there a way how to use javacv with ffmpeg and other libraries from the system? I'm a bit confused about the precompiled ffmpeg libraries as it is not clear to me what libraries are currently supported and if https://github.com/strukturag/libheif is actually used in the bytedeco release or not.

I've tried to search for the solution online, but apparently my prompting skills are not sufficient.

Thanks for help!

[IMG_2991.zip](https://github.com/user-attachments/files/17775967/IMG_2991.zip)

**Comment by Draczech** [comment]
**Update**
The code is working with following dependencies:

``` groovy
  bytedecoJavaCvVersion = '1.5.11-SNAPSHOT'
  bytedecoOpenCvVersion = "4.10.0-${bytedecoJavaCvVersion}"
  openBlasVersion = "0.3.28-${bytedecoJavaCvVersion}"
  ffmpegVersion = "7.1-${bytedecoJavaCvVersion}"
```

Maybe the ffmpeg version used by bytedeco is too old so it does not contain https://patchwork.ffmpeg.org/project/ffmpeg/patch/20231004164019.1428483-1-vigneshv@google.com/ or similar fix which is available in ffmpeg 7.1+

**Comment by saudet** [comment]
I've just released version 1.5.11 with FFmpeg 7.1, so yes please use that if it works.
